k2/python/k2/fsa_algo.py

```diff
@@ -36,6 +37,38 @@ def linear_fsa(symbols: Union[List[int], List[List[int]]]) -> Fsa:
     return fsa
 
 
+def linear_fst(symbols: Union[List[int], List[List[int]]],
+               aux_symbols: Union[List[int], List[List[int]]]) -> Fsa:
+    '''Construct an linear FST from symbols and it's corresponding
+       auxiliary symbols.
+
+    Note:
+      The scores of arcs in the returned FST are all 0.
+
+    Args:
+      symbols:
+        A list of integers or a list of list of integers.
+      aux_symbols:
+        A list of integers or a list of list of integers.
+
+    Returns:
+      An FST if the input is a list of integers.
+      A vector of FSTs if the input is a list of list of integers.
+    '''
+    ragged_arc = _k2.linear_fsa(symbols)
+    if isinstance(symbols[0], List):
+        assert isinstance(aux_symbols[0],
+                          List), 'aux_symbols and symbols does not match.'
+        for sym in aux_symbols:
+            sym.append(-1)  # -1 == kFinalSymbol
+        aux_labels = torch.IntTensor(aux_symbols).reshape(-1,)
```

Comment:
Does `torch.IntTensor` handle the case when each sublist in `aux_symbols` has a
different length?

Reply:
fixed it.

Reply:
I think it should be k2.RaggedInt, no?  
also rename all symbols->labels.

Reply:
Currently `compose` function seems to support `torch.tensor` only, see
https://github.com/k2-fsa/k2/blob/bbeb3ea65b1bfe9aec6ba62a06d42a6c78ad19a0/k2/python/k2/fsa_algo.py#L209

Reply:
Even if torch.IntTensor handles that, which I don't think it does, our code internally does not, it expects our own ragged tensor.

Reply:
That is specific to 'compose'.  The FSA format in general supports aux_labels being a k2.RaggedInt.

Reply:
So if I use `_k2.RaggedInt`, it would not work with `compose`, is my understanding right?

Reply:
> I think it should be k2.RaggedInt, no?

Currently, `RaggedInt` has only two constructors available in Python:
```
    1. _k2.RaggedInt(arg0: str)
    2. _k2.RaggedInt(shape: k2::RaggedShape, values: at::Tensor)
```

Shall we add another constructor supporting `List[List[int]]` ?

Reply:
I can take this task 😊